internal/download/download.go

```diff
@@ -115,50 +118,116 @@ func DownloadFile(url, outPath, header, token string) error {
 	return nil
 }
 
+func documentHashKey(document workflow.SpeakeasyRegistryDocument) string {
+	builder := strings.Builder{}
+	builder.WriteString(document.WorkspaceSlug)
+	builder.WriteString(":")
+	builder.WriteString(document.OrganizationSlug)
+	builder.WriteString(":")
+	builder.WriteString(document.NamespaceName)
+	builder.WriteString(":")
+	builder.WriteString(document.NamespaceID)
+	builder.WriteString(":")
+	builder.WriteString(document.Reference)
+	return builder.String()
+}
+
+func canCache(document workflow.SpeakeasyRegistryDocument) bool {
+	return strings.HasPrefix(document.Reference, "sha256:")
```

Comment:
This makes things kinda niche right? How often are images referenced directly with `sha256:`

Reply:
Every time we are looking up via a hash directly. This happens when:
1. We are calculating a change report based on workflow.lock
2. We are in e2e / local dev testing using the frozen-workflow-lockfile arg.

Unfortunately, we can't cache a tag `main` lookup (we could cache the blob, but not the manifest lookup) as tags can change under our feet.